### ISSUE TITLE
Restrict Content to Country

### ISSUE THREAD

**Issue by maverick74** [TeamNewPipe/NewPipe#8523]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Would be great to be able to restrict content to media produced by country.

### Why do you want this feature?

Kids (and also grown ups) spend a lot of time consuming content. Would be good to be able to restrict content to country .

In Portugal and Brazil the language is Portuguese but they are a lot different.

As such kids from both countries end up speaking the other country language.

This is creating a lot of school problems on both sides because there isn't a way to filter the content.

### Additional information

This would great and would prevent a lot of education problems

**Comment by litetex** [comment]
Might be fixed with https://github.com/TeamNewPipe/NewPipe/issues/2251, however no guarantee that this includes country/language related searches.

Anyway I don't think we can restrict the origin country in the first place. I can also upload a english speaking video to e.g. YT despite being German and it will be findable by US users despite having content country set to US, so I don't think there is a way of avoiding that...
I'm not sure if YT or any other service provides or can provide the country information for that.

I also don't see a problem why you shouldn't watch educational videos from e.g. Brazil when located in Portugal, science should work the same no matter where you are 😉 

And if you are searching for certain country specific stuff you should also include this in your searches. Example: You are located in Australia and want to search for some Australian law specific videos. Well then you should enter something like "Australian law" into the search field otherwise you will most likely get videos related to US or UK laws.

NewPipe also already provides an "content country"-option for Brazil and Portugal.

**Comment by maverick74** [comment]
> Might be fixed with #2251, however no guarantee that this includes country/language related searches.

Let's hope so!!! :)

> Anyway I don't think we can restrict the origin country in the first place. I can also upload a english speaking video to e.g. YT despite being German and it will be findable by US users despite having content country set to US, so I don't think there is a way of avoiding that... I'm not sure if YT or any other service provides or can provide the country information for that.
> 
> I also don't see a problem why you shouldn't watch educational videos from e.g. Brazil when located in Portugal, science should work the same no matter where you are 😉

This would be a choice we could enable or disable. As there would be no problem if we get a Portuguese video made in Brazil but with Portuguese-Portugal speaking language.

And yes, you are right for science there is on problem.

But when you go into a Portuguese class and start speaking Brazilian I won't even tell you where your grades will end up.
I'm REALLY not joking! It's really a very serious problem!!! While you do write sort Of The same, you do speak very differently!

> And if you are searching for certain country specific stuff you should also include this in your searches. Example: You are located in Australia and want to search for some Australian law specific videos. Well then you should enter something like "Australian law" into the search field otherwise you will most likely get videos related to US or UK laws.

Usually people use search term + "pt br" or "pt pt" or "Portuguese Brazil" or "Portuguese Portugal" to try to get Better results

> NewPipe also already provides an "content country"-option for Brazil and Portugal.

I believe I already set all options to the right language... But maybe I missed something. Could you tell the path to this setting?

**Comment by maverick74** [comment]
> Might be fixed with #2251, however no guarantee that this includes country/language related searches.

Let's hope so!!! :)

> Anyway I don't think we can restrict the origin country in the first place. I can also upload a english speaking video to e.g. YT despite being German and it will be findable by US users despite having content country set to US, so I don't think there is a way of avoiding that... I'm not sure if YT or any other service provides or can provide the country information for that.
> 
> I also don't see a problem why you shouldn't watch educational videos from e.g. Brazil when located in Portugal, science should work the same no matter where you are 😉

This would be a choice we could enable or disable. As there would be no problem if we get a Portuguese video made in Brazil but with Portuguese-Portugal speaking language.

And yes, you are right for science there is on problem.

But when you go into a Portuguese class and start speaking Brazilian I won't even tell you where your grades will end up.
I'm REALLY not joking! It's really a very serious problem!!! While you do write sort Of The same, you do speak very differently!

> And if you are searching for certain country specific stuff you should also include this in your searches. Example: You are located in Australia and want to search for some Australian law specific videos. Well then you should enter something like "Australian law" into the search field otherwise you will most likely get videos related to US or UK laws.

Usually people use search term + "pt br" or "pt pt" or "Portuguese Brazil" or "Portuguese Portugal" to try to get Better results

> NewPipe also already provides an "content country"-option for Brazil and Portugal.

I believe I already set all options to the right language... But maybe I missed something. Could you tell the path to this setting?

**Comment by SameenAhnaf** [comment]
>NewPipe also already provides an "content country"-option for Brazil and Portugal.

`Settings`>`Content`>`Default Content Country`. Choose your country.

**Comment by maverick74** [comment]
> It should have been applied for related streams too.

But is that a YT limitiation, or a NP limitation?

**Comment by litetex** [comment]
> It should have been applied for related streams too.

According to the extractor code the country is already applied to requests. However YouTube's algorithms may still server different language content.

<details><summary>Kurzgesagt demo</summary>

![grafik](https://user-images.githubusercontent.com/40789489/174893823-723fe096-b4bf-48c5-8365-dd20f816d75c.png)

</details>

And I think there is nothing we can do...

**Comment by opusforlife2** [comment]
Youtube Help says:
> The location you choose impacts the videos that surface for:
    Recommendations
    Trending
    News

**Comment by wtusg** [comment]
Although this is not entirely related to the problem, but I want to say that I think it would be convenient if on the first launch NewPipe see, for example, the Portuguese language in the phone settings, a pop-up menu would allow user to select a proper region.

**Comment by opusforlife2** [comment]
People in the know: are "recommendations" different from "related videos"?

**Comment by carstentrink** [comment]
> Kids (and also grown ups) spend a lot of time consuming content. Would be good to be able to restrict content to country .

The official YouTube app also cannot do this.

> As such kids from both countries end up speaking the other country language.

Surprising to know. I never see American speaking British English (i.e) English!!!

> This is creating a lot of school problems on both sides because there isn't a way to filter the content.

I would have thought, it would be better to broaden knowledge etc.

**Comment by maverick74** [comment]
Yeah, YT does not do that. That's why i asked for NP to add the feature, if it's technically possible... (and i do know it's a looooong shot)

> Surprising to know. I never see American speaking British English (i.e) English!!!

Yeah, it's really funny (except no one laughing :). 
You can do an easy test: Get to facebook and find some portuguese (pt-pt) parents and ask them if the situation applies. Also ask if anyone is worried about it.

> I would have thought, it would be better to broaden knowledge etc.

Well, there are two things with that statement:
1. If everything was about spreading knowledge the problem would be a bit smaller, but, then again, you would still have portugueses speaking brazilian and creating problems at schools as well as having problems with mixing culture and culture-language specific problems (that's apparently the reason why france made some new laws to protect french language)


2. however, has hinted in 1. this is not just about spreading knowledge. The majority of kids  are consuming junk-media instead of "knowledge"... so " broaden knowledge" does not quite applies (beside, everyone understands brazilian around here. If you want to  broaden knowledge in language then they should try english - which is kind of "universal" and will indeed be very useful in their future)

**Comment by maverick74** [comment]
> Closing in favour of [#2251](https://github.com/TeamNewPipe/NewPipe/issues/2251)

I don't really agree as it seem to have different scopes....but as long as I can block Brazilian videos from coming up I'll be happy with the feature

**Comment by maverick74** [comment]
I would like to say that for many around here it is **brutally frustrating** to want _portuguese (Portugal) - ONLY_ videos and keep getting Brazilian videos popping up all the time in the list.

It could be just me... but unfortunately everyone complains the same problem which google - stubbornly - does not fix!!! (some say it feels _they have an agenda_ or something, and i'm starting to agree...)
